modules/data.atmosphere/R/download.NOAA_GEFS_downscale.R

```diff
@@ -0,0 +1,341 @@
+  
+  #Check to see if start_date is valid. This must be done after date adjustment.
+  if (as.POSIXct(Sys.time(), tz="UTC") < start_date || start_date < NOAA_GEFS_Start_Date) {
+    PEcAn.logger::logger.severe(sprintf('Start date (%s) exceeds the NOAA GEFS range (%s to %s).',
+                                        start_date,
+                                        NOAA_GEFS_Start_Date, Sys.Date()))
+  }
+  
+  if (lubridate::hour(start_date) > 23) {
+    PEcAn.logger::logger.severe(sprintf("Start time %s is not a valid time", lubridate::hour(start_date)))
+  }
+  
+  if (lubridate::hour(end_date) > 23) {  #Done separately from the previous if statement in order to have more specific error messages.
+    PEcAn.logger::logger.severe(sprintf("End time %s is not a valid time", lubridate::hour(end_date)))
+  }
+  #End date/time error checking
+  
+  #################################################
+  #NOAA variable downloading
+  #Uses the rnoaa package to download data
+  
+  #We want data for each of the following variables. Here, we're just getting the raw data; later, we will convert it to the 
+  #cf standard format when relevant.
+  noaa_var_names = c("Temperature_height_above_ground_ens", "Pressure_surface_ens", "Relative_humidity_height_above_ground_ens", "Downward_Long-Wave_Radp_Flux_surface_6_Hour_Average_ens", 
+                     "Downward_Short-Wave_Radiation_Flux_surface_6_Hour_Average_ens", "Total_precipitation_surface_6_Hour_Accumulation_ens",
+                     "u-component_of_wind_height_above_ground_ens", "v-component_of_wind_height_above_ground_ens")
+  
+  cf_var_names = c("air_temperature", "air_pressure", "specific_humidity", "surface_downwelling_longwave_flux_in_air", 
+                   "surface_downwelling_shortwave_flux_in_air", "precipitation_flux", "eastward_wind", "northward_wind")
+  #These are the cf standard names
+  cf_var_names1 = c("air_temperature", "air_pressure", "specific_humidity", "surface_downwelling_longwave_flux_in_air", 
+                    "surface_downwelling_shortwave_flux_in_air", "precipitation_flux", "wind_speed")
+  cf_var_units = c("K", "Pa", "1", "Wm-2", "Wm-2", "kgm-2s-1", "ms-1")  #Negative numbers indicate negative exponents
+  
+  # This debugging loop allows you to check if the cf variables are correctly mapped to the equivalent
+  # NOAA variable names.  This is very important, as much of the processing below will be erroneous if 
+  # these fail to match up.
+  # for (i in 1:length(cf_var_names)) {
+  #  print(sprintf("cf / noaa   :   %s / %s", cf_var_names[[i]], noaa_var_names[[i]]))
+  #}
+  
+  noaa_data = list()
+  
+  #Downloading the data here.  It is stored in a matrix, where columns represent time in intervals of 6 hours, and rows represent
+  #each ensemble member.  Each variable getxs its own matrix, which is stored in the list noaa_data.
+  for (i in 1:length(noaa_var_names)) {
```

Comment:
`noaa_var_names` is hard coded here, so it shouldn't really matter, but strictly speaking, `seq_along` is safer.

```suggestion
  for (i in seq_along(noaa_var_names)) {
```